web-common/src/features/canvas/stores/canvas-spec.ts

```diff
@@ -66,6 +70,13 @@ export class CanvasResolvedSpec {
       Object.keys($validSpecStore?.data?.metricsViews || {}),
     );
 
+    this.getMetricsViewFromName = (metricViewName: string) =>
```

Comment:
Nit: I'd rename to `getMetricsViewValidSpecFromName`, since it returns a `validSpec`, not the whole `metricsView` resource